### PR TITLE
[Merged by Bors] - feat: support project secret on api-sdk (PL-296)

### PR DESCRIPTION
**Pull request by pmvrmc**
**Fixes or implements PL-296**

### Brief description. What is this change?
Adds new project secret functionality to the api-sdk.

Needs to be updated with package.json changes after the base-types PR is merged (we use a new type in this PR)

### Related PRs

- https://github.com/voiceflow/libs/pull/392
- https://github.com/voiceflow/database-cli/pull/295
- https://github.com/voiceflow/platform/pull/309
- https://github.com/voiceflow/general-runtime/pull/448

### Checklist

- [ ] changes have been validated in an ephemeral environment
- [ ] this is a breaking change and should publish a new major version
- [ ] appropriate tests have been written
- [ ] any new env vars have been added to the [notion doc](https://www.notion.so/voiceflow/Add-Environment-Variables-be1b0136479f45f1adece7995a7adbfb) and infra has been notified

### CODE REVIEW

File: packages/api-sdk/src/resources/projectSecret.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    return data;
+  }
+
+  public async find(lookup: string, tag: number): Promise<BaseModels.ProjectSecret.Model[]> {
+    const { data } = await this.fetch.get<BaseModels.ProjectSecret.Model[]>(
+      `${this.endpoint}/secret?lookup=${encodeURIComponent(lookup)}&tag=${tag}`
+    );
+
+    return data;
+  }
```
Comment (AhmedVF) [inline packages/api-sdk/src/resources/projectSecret.ts:25]: Just wondering, do you think it's possible to better differentiate these two functions using their names?
Maybe `findByProjectID` vs `findByLookup` or something.

Comment (pmvrmc) [reply packages/api-sdk/src/resources/projectSecret.ts:25]: sure, will use that suggestion

File: packages/api-sdk/src/resources/projectSecret.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+  public async findByLookup(lookup: string, tag: number): Promise<BaseModels.ProjectSecret.Model[]> {
+    const { data } = await this.fetch.get<BaseModels.ProjectSecret.Model[]>(
+      `${this.endpoint}/secret?lookup=${encodeURIComponent(lookup)}&tag=${tag}`
+    );
+
+    return data;
+  }
```
Comment (AhmedVF) [inline packages/api-sdk/src/resources/projectSecret.ts:25]: I'm also curious, how come for one of these, you return only the secret, but for the other, you return the entire ProjectSecret object?

Comment (pmvrmc) [reply packages/api-sdk/src/resources/projectSecret.ts:25]: you are right, nice catch!

**Review by AhmedVF — APPROVED** [review]
LGTM

**Pull request closed without merging**